cmd/api/notifications/notifications.go

```diff
@@ -6,37 +6,33 @@ import (
 	"fmt"
 	"net/http"
 	"strings"
-	"time"
 )
 
 type Ntfy struct {
 	baseURL string
 	enabled bool
-	timeout time.Duration
 	client  *http.Client
 }
 
-func NewNtfy(enableNotifications bool, notificationsTimeout time.Duration, notificationsBaseURL string) *Ntfy {
+func NewNtfy(enableNotifications bool, notificationsBaseURL string) *Ntfy {
 	return &Ntfy{
 		baseURL: notificationsBaseURL,
 		enabled: enableNotifications,
-		timeout: notificationsTimeout,
 		client:  &http.Client{},
 	}
 }
 
-func (ntf *Ntfy) BookCreated(title string, inventory int) error {
+func (ntf *Ntfy) BookCreated(ctx context.Context, title string, inventory int) error {
 	if ntf.enabled {
-		ctx, cancel := context.WithTimeout(context.Background(), ntf.timeout)
-		defer cancel()
-		req, err := http.NewRequestWithContext(ctx, http.MethodPost, ntf.baseURL+"/New_book_created", strings.NewReader(fmt.Sprintf("New book created:\nTitle: %s\nInventory: %v", title, inventory)))
+		req, err := http.NewRequestWithContext(ctx, http.MethodPost, ntf.baseURL+"_New_book_created", strings.NewReader(fmt.Sprintf("New book created:\nTitle: %s\nInventory: %v", title, inventory))) //Ntfy SEEMS NOT TO ACEPT SLASHS OR DOTS AT TOPIC
 		if err != nil {
-			return fmt.Errorf("error delivering message ("+fmt.Sprintf("New book created:\nTitle: %s\nInventory: %v", title, inventory)+") to topic ("+ntf.baseURL+"/New_book_created): %w", err)
+			return fmt.Errorf("error delivering message ("+fmt.Sprintf("New book created: Title: %s Inventory: %v", title, inventory)+"): %w", err)
 		}
 		_, err = ntf.client.Do(req)
 		if err != nil {
-			return fmt.Errorf("error delivering message ("+fmt.Sprintf("New book created:\nTitle: %s\nInventory: %v", title, inventory)+") to topic ("+ntf.baseURL+"/New_book_created): %w", err)
+			return fmt.Errorf("error delivering message ("+fmt.Sprintf("New book created: Title: %s Inventory: %v", title, inventory)+"): %w", err)
 		}
+		return nil
 	}
 	return errors.New("notifications not enabled")
 }
```

Comment:
With code the way it is, after we send the notification, we are not listening to the topic to certify that the message was delivered. We could do this with a GET request, like I did in notifications_test.go. Do you think its necessary?

Reply:
This is the publisher of the message, you don't need to assert that the message was read anywhere.

Reply:
ok